memtagabielf64/memtagabielf64.rst

```diff
@@ -366,9 +366,9 @@ MemtagABI adds the following processor-specific dynamic array tags:
   +---------------------------------+------------+--------+-------------------+-------------------+
   | ``DT_AARCH64_MEMTAG_HEAP``      | 0x7000000b | d\_val | Platform specific | Platform Specific |
   +---------------------------------+------------+--------+-------------------+-------------------+
-  | ``DT_AARCH64_MEMTAG_STACK``     | 0x7000000c | d\_ptr | Platform specific | Platform Specific |
+  | ``DT_AARCH64_MEMTAG_STACK``     | 0x7000000c | d\_val | Platform specific | Platform Specific |
```

Comment:
The ELF rules for d_val and d_ptr for tags in the LOOS and HIOS range are inferred from whether the tag is even or odd.

As the spec is in Alpha I'm assuming ABI breaks are possible? If so I recommend switching the order of the tags in the table.

I'm not sure quite what to do if ABI breaks are not possible. We may need to add more tags with the right even/odd value so that if this ever got deplyed outside of Android it could be used by a dynamic loader that inferred val/ptr from that.

From https://www.sco.com/developers/gabi/latest/ch5.dynamic.html
```
To make it simpler for tools to interpret the contents of dynamic section entries, the value of each tag, except for those in two special compatibility ranges, will determine the interpretation of the d_un union. A tag whose value is an even number indicates a dynamic section entry that uses d_ptr. A tag whose value is an odd number indicates a dynamic section entry that uses d_val or that uses neither d_ptr nor d_val. Tags whose values are less than the special value DT_ENCODING and tags whose values fall between DT_HIOS and DT_LOPROC do not follow these rules.
```

Reply:
Thanks for the update I'll need to update my pull request to reserve the numbers in the main specification https://github.com/ARM-software/abi-aa/pull/216

Reply:
(resolving, looks like #216 doesn't have the new types)